test/OpenTelemetry.Contrib.Shared.Benchmarks/SqlProcessorBenchmarks.cs

```diff
@@ -0,0 +1,141 @@
+
+// * Hints *
+Outliers
+  SqlProcessorBenchmarks.Simple: Default -> 1 outlier  was  removed (410.19 ns)
+  SqlProcessorBenchmarks.Simple: Default -> 1 outlier  was  removed (917.40 ns)
+
+
+AFTER SPAN APPEND TARGET + ALLOCATE STRINGBUILDER WITH CAPACITY (CACHE ENABLED)
+-------------------------------------------------------------------------------
+BenchmarkDotNet v0.13.12, Windows 11 (10.0.26100.4946)
+Unknown processor
+.NET SDK 9.0.304
+  [Host]     : .NET 9.0.8 (9.0.825.36511), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
+  DefaultJob : .NET 9.0.8 (9.0.825.36511), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
+
+
+| Method | Sql                  | Iterations | Mean     | Error    | StdDev   | Allocated |
+|------- |--------------------- |----------- |---------:|---------:|---------:|----------:|
+| Simple | SELEC(...)ls od [39] | 1          | 29.11 ns | 0.586 ns | 0.549 ns |         - |
+| Simple | SELE(...)_id) [101]  | 1          | 67.52 ns | 0.987 ns | 0.923 ns |         - |
+
+// * Hints *
+Outliers
+  SqlProcessorBenchmarks.Simple: Default -> 1 outlier  was  removed (35.26 ns)
+
+
+AFTER SWITCHING TO DICTIONARY CACHE (DISABLED)
+----------------------------------------------
+BenchmarkDotNet v0.13.12, Windows 11 (10.0.26100.4946)
+Unknown processor
+.NET SDK 9.0.304
+  [Host]     : .NET 9.0.8 (9.0.825.36511), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
+  DefaultJob : .NET 9.0.8 (9.0.825.36511), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
+
+
+| Method | Sql                  | Iterations | Mean     | Error    | StdDev   | Gen0   | Allocated |
+|------- |--------------------- |----------- |---------:|---------:|---------:|-------:|----------:|
+| Simple | SELEC(...)ls od [39] | 1          | 355.9 ns |  3.49 ns |  2.92 ns | 0.0420 |     528 B |
+| Simple | SELE(...)_id) [101]  | 1          | 804.1 ns | 14.63 ns | 12.97 ns | 0.0725 |     912 B |
+
+// * Hints *
+Outliers
+  SqlProcessorBenchmarks.Simple: Default -> 2 outliers were removed (367.28 ns, 370.58 ns)
+  SqlProcessorBenchmarks.Simple: Default -> 2 outliers were removed (860.17 ns, 906.11 ns)
+
+
+AFTER SWITCHING TO DICTIONARY CACHE (ENABLED)
+---------------------------------------------
+BenchmarkDotNet v0.13.12, Windows 11 (10.0.26100.4946)
+Unknown processor
+.NET SDK 9.0.304
+  [Host]     : .NET 9.0.8 (9.0.825.36511), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
+  DefaultJob : .NET 9.0.8 (9.0.825.36511), X64 RyuJIT AVX-512F+CD+BW+DQ+VL+VBMI
+
+
+| Method | Sql                  | Iterations | Mean     | Error    | StdDev   | Allocated |
+|------- |--------------------- |----------- |---------:|---------:|---------:|----------:|
+| Simple | SELEC(...)ls od [39] | 1          | 15.47 ns | 0.336 ns | 0.373 ns |         - |
+| Simple | SELE(...)_id) [101]  | 1          | 30.94 ns | 0.505 ns | 0.472 ns |         - |
+
+// * Hints *
+Outliers
+  SqlProcessorBenchmarks.Simple: Default -> 1 outlier  was  removed (18.57 ns)
+*/
```

Comment:
Maybe check the `.md` result files in initially with the old implementation, then another commit with the updated ones? Then they're easy to read using the Git diff (basically what I did just now with https://github.com/App-vNext/Polly/commit/cf50bc52196f3153ca8ca7b2f6900d8d451a4a92).

Reply:
Yeah, can do. I saw that in most other benchmarks one or more results where just commented at the top, so I copied that. We don't need the full trail here though.

Reply:
@martincostello , I've updated the config and commited the before and after results.